tests/pyfunc/test_warn_dependency_requirement_mismatches.py

```diff
@@ -110,3 +111,30 @@ def mock_get_installed_version_fn(package, module=None):
                     "detecting model dependency mismatches"
                 )
             )
+
+
+@pytest.mark.large
+def test_warn_dependency_requirement_mismatches_ignore_databricks_runtime_micro_version(tmpdir):
+    req_file = tmpdir.join("requirements.txt")
+    req_file.write("pyspark==3.2.1")
+
+    with mock.patch("mlflow.pyfunc._logger.warning") as mock_warning:
+        with mock.patch("mlflow.utils.requirements_utils.is_in_databricks_runtime", lambda: True):
+            for pyspark_version in ["3.2.1", "3.2.2"]:
+                with mock.patch(
+                    "mlflow.utils.requirements_utils._get_installed_version",
+                    _gen_mock_get_installed_version_fn({"pyspark": pyspark_version}),
+                ):
+                    _warn_dependency_requirement_mismatches(model_path=tmpdir)
+                    mock_warning.assert_not_called()
+                    mock_warning.reset_mock()
+
+        with mock.patch(
+            "mlflow.utils.requirements_utils._get_installed_version",
+            _gen_mock_get_installed_version_fn({"pyspark": "3.2.2"}),
+        ):
+            _warn_dependency_requirement_mismatches(model_path=tmpdir)
+            mock_warning.assert_called_once_with(
+                AnyStringWith(" - pyspark (current: 3.2.2, required: pyspark==3.2.1)")
+            )
+            mock_warning.reset_mock()
```

Comment:
`3.2.2` and `3.2.1` should be considered as matched, right?

Reply:
this is test for non-databricks pyspark.